short/index.html

```diff
@@ -298,6 +298,35 @@
 </section>
 
 <section data-markdown>
+**Exercise**
+
+```python
+def test_overlay_no_overlap():
+    rect1 = Rectangle(0, 0, 1, 1)
+    rect2 = Rectangle(2, 0, 2, 2)
+    assert overlay(rect1, rect2) == None
+
+def test_overlay_contains():
+    rect1 = Rectangle(0, 0, 5, 5)
+    rect2 = Rectangle(2, 2, 3, 3)
+    rect_ov = overlay(rect1, rect2)
+
+    assert rect_ov.x0 == 0
+    assert rect_ov.y0 == 0
+    assert rect_ov.x1 == 5
+    assert rect_ov.y1 == 5
+
+def test_overlay_area():
+    rect1 = Rectangle(0, 0, 5, 5)
+    rect2 = Rectangle(0, 1, 5, 2)
+    rect_ov = overlay(rect1, rect2)
+
+    assert rect_ov.area() <= rect1.area()
+    assert rect_ov.area() <= rect2.area()
```

Comment:
> I think it's really important to have tests for rectangles that overlap on an edge and at a point - corner cases (literally).

This was brought up by attendees in both meetings, and I was prepared to mention these cases if nobody else did. This slide was just showing some examples, but I agree the edge cases are more interesting/important than the area test.